neurokit2/rsp/rsp_rate.py

```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
+import numpy as np
+
+from ..signal import (signal_resample, signal_rate, signal_findpeaks, signal_interpolate,
+                      signal_filter)
+from .rsp_peaks import rsp_peaks
+
+from matplotlib import pyplot as plt
+import scipy.stats
+
+
+def rsp_rate(rsp_cleaned, peaks=None, sampling_rate=1000, window=10, hop_size=1, method="peak",
+             peak_method="khodadad2018"):
+    """Find respiration rate by cross-correlation method.
+    Parameters
+    ----------
+    rsp_cleaned : Union[list, np.array, pd.Series]
+        The cleaned respiration channel as returned by `rsp_clean()`.
+    sampling_rate : int
+        The sampling frequency of 'rsp_cleaned' (in Hz, i.e., samples/second).
+    window : int
+        The duration of the sliding window (in second). Default to 10 seconds.
+    hop_size : int
+        The number of samples between each successive window. Default to 1 sample.
+    method : str
+        Method can be either "peak" or "xcorr". In "peak" method, rsp_rate is calculated from the
+        periods between respiration peaks. In "xcorr" method, cross-correlations between the changes
+        in respiration with a bank of sinusoids of different frequencies are caclulated to indentify
+        the principal frequency of oscillation.
+    peak_method : str
+        Method to identify respiration peaks. Can be one of "khodadad2018" (default) or "biosppy".
+
+    Return
+    ------
+    rsp_rate : np.ndarray
+        Instantenous respiration rate.
+
+    Example
+    -------
+    >>> import neurokit2 as nk
+    >>> rsp_signal = nk.rsp_simulate(100, respiratory_rate=15)
+    >>> rsp_cleaned = nk.rsp_clean(rsp_signal)
+    >>>
+    >>> rsp_rate = nk.rsp_rate(rsp_cleaned, method="peaks")
+    >>> rsp_rate2 = nk.rsp_rate(rsp_cleaned, method="xcorr")
+    """
+
+    if method.lower() in ["peak", "peaks", "signal_rate"]:
+        if peaks is None:
+            peaks, info = rsp_peaks(rsp_cleaned, sampling_rate=sampling_rate, method=peak_method)
+        rsp_rate = signal_rate(peaks, sampling_rate=sampling_rate, desired_length=len(rsp_cleaned))
+
+    if method.lower() in ["cross-correlation", "xcorr"]:
```

Comment:
I'd also probably do an if, elif, else error "wrong method" flow to throw informative errors